components/nodemanager-service/pgdb/node-managers-process-node.go

```diff
@@ -86,9 +168,47 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
 		}
 		return tx.tagNode(node.GetUuid(), tags)
 	})
+
+	return err
+}
+
+func (db *DB) getCurrentScanStatus(node *manager.NodeMetadata) (string, error) {
```

Comment:
i played around with the idea of doing away with this function and doing it all in postgres, but it didn't work well for me, as we overwrite the data later when we insert the run_data or scan_data. 
But if ppl know of a way to do this and not run into that problem, i'm all ears!

Reply:
What did you try, and what happened? I'd have to play around with it a bit but I think you should be able to access the current value for the field during the upsert and set the appropriate field with it.

Reply:
I was grabbing the value of status, setting penultimate_status to that value, and then going on with the rest of it, but it seemed like it was getting overwritten.
i'll play around again today with it. seems like it should be doable and it would simplify this logic alot

Reply:
hmmm. ok, i've tried revisiting this. first i was trying to just set the value of penultimate status to current value of status. that was not working as the intent of json in postgres is to store blurbs of data, not really made for updating individual fields within that data well.
so then i thought i could make an extra couple columns for just penult status for scan and run. the problem is updating any of these fields from within the queries is super awkward (i can't get it to work right) b/c they are upserts...